components/script/dom/promise.rs

```diff
@@ -61,30 +94,17 @@ impl PromiseHelper for Rc<Promise> {
     #[allow(unsafe_code)]
     fn initialize(&self, cx: SafeJSContext) {
         let obj = self.reflector().get_jsobject();
-        self.permanent_js_root.set(ObjectValue(*obj));
+        self.set_permanent_js_root(ObjectValue(*obj));
         unsafe {
             assert!(AddRawValueRoot(
                 *cx,
-                self.permanent_js_root.get_unsafe(),
+                self.get_unsafe_permanent_js_root(),
                 c"Promise::root".as_ptr(),
             ));
         }
     }
 }
 
-impl Drop for Promise {
-    #[allow(unsafe_code)]
```

Comment:
This one is actually a special case. Promise objects are stored inside Rc values, so the Drop is run when the last Rc is dropped, rather than when SpiderMonkey runs a GC. We should leave this file alone.

Reply:
So should we close this PR and leave Promise as they are?

Reply:
Let's add a comment to the existing Drop code that says:
```
Promise objects are stored inside Rc values, so Drop is run when the last Rc is dropped, rather than when SpiderMonkey runs a GC. This makes it safe to interact with the JS engine unlike Drop implementations for other DOM types.
```

Reply:
I will add the comment in another PR